core/database/arango_db_manager.py

```
@@ -0,0 +1,252 @@
+                    if attempt < max_retries:
+                        # Add jitter to prevent thundering herd
+                        jittered_delay = delay * (0.5 + random.random())
+                        logger.warning(
+                            f"Database operation '{func.__name__}' failed (attempt {attempt + 1}/{max_retries + 1}): {type(e).__name__}: {e}. "
+                            f"Retrying in {jittered_delay:.2f}s..."
+                        )
+                        time.sleep(jittered_delay)
+                        delay = min(delay * exponential_base, max_delay)
+                    else:
+                        raise
+                except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)